hal/src/rtl872x/exflash_hal.cpp

```diff
@@ -41,6 +41,10 @@ extern "C" {
 extern uintptr_t platform_system_part1_flash_start;
 extern uintptr_t platform_flash_end;
 
+static bool gXipControl = false;
```

Comment:
Let's keep all of this inside the class. If there is a static entry in the MPU table that we are modifying, just make it a static member variable. For initial disabling of XIP, just add a new method to initialize things etc and do it in hal_exflash_disable_xip

Reply:
Reworked